keystore/src/traits/entity_encryption_ext.rs

```diff
@@ -0,0 +1,78 @@
+const AES_GCM_256_NONCE_SIZE: usize = 12;
+
+#[derive(core_crypto_macros::Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+struct Aad {
+    type_name: Vec<u8>,
+    id: Vec<u8>,
+}
+
+/// Entity-level encryption extensions
+///
+/// This trait should be removed once we have moved away from the IDB backend
+/// into some (hopefully unified) database backend which supports database-level
+/// encryption at rest. See WPB-16241.
+#[async_trait::async_trait(?Send)]
+pub trait EntityEncryptionExt: Entity {
+    fn aad(&self) -> CryptoKeystoreResult<Vec<u8>> {
+        let aad = Aad {
+            type_name: Self::COLLECTION_NAME.as_bytes().to_vec(),
+            id: self.id_raw().into(),
+        };
+        serde_json::to_vec(&aad).map_err(Into::into)
+    }
+
+    // About WASM Encryption:
+    // The store key (i.e. passphrase) is hashed using SHA256 to obtain 32 bytes
```

Comment:
This is not the case anymore, since the database key type change clients have to provide the key directly .